src/registry/persistent.rs

```diff
@@ -0,0 +1,58 @@
+//! Defines a persistent memory based [Registry].
+
+use std::fs::File;
+use std::io::{self, Read, Write};
+
+use sha2::{Digest, Sha256};
+
+use super::{ContentHash, Registry};
+
+/// A persistent memory based [Registry].
+pub struct PersistentRegistry {
+    /// The path to the folder where the [Registry] is stored.
+    path: String,
+}
+
+impl Registry for PersistentRegistry {
+    type Error = io::Error;
+
+    fn read(&self, hash: ContentHash) -> Result<Option<impl Read>, Self::Error> {
+        let path = format!("{}/{}", self.path, hash);
+
+        let file = match File::open(path) {
+            Ok(file) => file,
+            Err(err) => return Err(err),
+        };
+
+        Ok(Some(file))
+    }
+
+    fn write(&self, mut content: impl Read) -> Result<ContentHash, Self::Error> {
+        let mut hasher = Sha256::new();
+
+        // Create a temporary file to store the content in.
+        let mut temp_file = File::create(format!("{}/{}", self.path, "temp"))?;
```

Comment:
This will not work with concurrent writing, you can use a Uuid to get a unique name.